src/routes.php

```diff
@@ -83,11 +83,11 @@
 
 // Import route
 $app->post('/run/import', function () use ($di, $app) {
-    $app->controller = $di['importController'];
-    $app->controller->import();
+    /** @var Xhgui_Controller_Import $controller */
+    $controller = $di['importController'];
```

Comment:
Probably my bad :)

I looked at the surrounding bits and didn't fully understand first how/why it was the way it was.

There is predecent for not setting `$app->controller`, such as:

```php
$app->post('/run/delete_all', function () use ($di, $app) {
    $di['runController']->deleteAllSubmit();
})->name('run.deleteAll.submit');
```

Reply:
I found that the response is the error by accident. the saver does error code of submit at all :)

And the fix is accidental too, I saw some code that $app->controller is related to template rendering. I'm not familiar with the framework either.

Reply:
@Krinkle, the /run/import was added in https://github.com/perftools/xhgui/pull/230

how are you related? :)